packages/server/tests/acceptance/erc20.spec.ts

```diff
@@ -0,0 +1,295 @@
+                                        `CALL_EXCEPTION`,
+                                    );
+                                });
+                            });
+
+                            describe('when the token owner does not have enough balance', function () {
+                                let amount;
+                                before(async function () {
+                                    amount = allowance;
+                                });
+
+                                beforeEach('reducing balance', async function () {
+                                    await contract.transfer(to, 2, { from: tokenOwner });
+                                });
+
+                                it('reverts', async function () {
+                                    await expectRevert(
+                                        contract.connect(spenderWallet).transferFrom(tokenOwner, to, amount),
+                                        `CALL_EXCEPTION`,
+                                    );
+                                });
+                            });
+                        });
+
+                        describe('when the spender has unlimited allowance', function () {
+                            beforeEach(async function () {
+                                await contract.connect(tokenOwnerWallet).approve(spender, ethers.constants.MaxUint256);
+                            });
+
+                            it('does not decrease the spender allowance', async function () {
+                                await contract.connect(spenderWallet).transferFrom(tokenOwner, to, 1);
+                                const allowance = await contract.allowance(tokenOwner, spender);
+                                expect(allowance.toString()).to.be.equal(ethers.constants.MaxUint256.toString());
+                            });
+
+                            it('does not emit an approval event', async function () {
+                                await expect(await contract.connect(spenderWallet).transferFrom(tokenOwner, to, 1))
+                                    .to.not.emit(contract, 'Approval');
+                            });
+                        });
+                    });
+
+                    describe('when the recipient is the zero address', function () {
+                        let amount, to, tokenOwnerWallet;
+
+                        beforeEach(async function () {
+                            amount = initialSupply;
+                            to = ethers.constants.AddressZero;
+                            tokenOwnerWallet = accounts[0].wallet;
+                            await contract.connect(tokenOwnerWallet).approve(spender, amount);
+                        });
+
+                        it('reverts', async function () {
+                            await expectRevert(contract.connect(spenderWallet).transferFrom(tokenOwner, to, amount),
```

Comment:
Ideally we should make the revert messages be parsed correctly by client libraries. Logged it as a separate issue:
https://github.com/hashgraph/hedera-json-rpc-relay/issues/279